src/my_board.rs

```diff
@@ -17,71 +17,48 @@ impl Status {
 
 #[derive(Debug, Clone)]
 pub struct MyBoard {
-    pub board: Vec<Status>,
-    pub size: usize,
+    pub board: [[CellType; 20]; 20],
+    pub size: u8,
 }
 
 struct Move {
-    move_position: usize,
+    x: u8,
+    y: u8,
     next_moves: Vec<Move>,
 }
 
 impl MyBoard {
     pub fn new() -> Self {
         MyBoard {
-            board: Vec::new(),
-            size: 0,
+            board: [[CellType::Empty; 20]; 20],
+            size: 20,
         }
     }
 
-    pub fn resize(&mut self, size: usize) {
-        let new_size = size * size;
-        self.board = vec![Status::Empty; new_size];
-    }
-
-    pub fn fetch_cell(&mut self, x: usize, y: usize) -> Status {
-        let cell = (y * self.size) + x;
-        self.board[cell]
-    }
-
-    pub fn set_cell(&mut self, x: usize, y: usize, state: Status) {
-        let cell = (y * self.size) + x;
-        self.board[cell] = state;
-    }
-
-    pub fn index_to_coordinates(
-        &self,
-        index: usize,
-    ) -> (usize, usize) {
-        let x = index % self.size;
-        let y = index / self.size;
-        (x, y)
-    }
-
     pub fn print(&self) {
-        for (i, status) in self.board.iter().enumerate() {
-            let symbol = match status {
-                Status::Empty => '.',
-                Status::Enemy => 'X',
-                Status::Ally => 'O',
-            };
-            print!("{} ", symbol);
-            if (i + 1) % self.size == 0 {
-                println!();
+        for y in self.board.iter() {
+            for x in y.iter() {
+                let symbol = match x {
+                    CellType::Empty => '.',
+                    CellType::Enemy => 'X',
+                    CellType::Ally => 'O',
+                };
+                print!("{} ", symbol);
             }
+            println!();
         }
     }
 
     pub fn clear_board(&mut self) {
-        for status in self.board.iter_mut() {
-            *status = Status::Empty;
-        }
+        self.board = [[CellType::Empty; 20]; 20];
     }
 
-    pub fn calculate_next_move(&self) -> (usize, usize) {
+    pub fn calculate_next_move(&self) -> (u8, u8) {
         let root = self.generate_tree();
         let mut best_move = &Move {
-            move_position: usize::MAX,
+            // I don't really know what values to put there, could you help me ?
```

Comment:
We'll think about that when implementing a proper decision tree

Reply:
Okay 🫡